api/src/main/java/io/opencensus/common/Functions.java

```diff
@@ -72,4 +80,17 @@ public T apply(Object ignored) {
     Function<Object, T> function = (Function<Object, T>) THROW_ILLEGAL_ARGUMENT_EXCEPTION;
     return function;
   }
+
+  /**
+   * A {@code Function} that always ignores its argument and throws an {@link AssertionError}.
+   *
+   * @return a {@code Function} that always ignores its argument and throws an {@code
+   *     AssertionError}.
+   */
+  public static <T> Function<Object, T> throwAssertionError() {
+    // It is safe to cast a producer of Void to anything, because Void is always null.
```

Comment:
I like explaining, but isn't this focusing on the wrong thing? ex you could have object result. the fact that assertion error is raised always negates the need to explain return type choices, right?